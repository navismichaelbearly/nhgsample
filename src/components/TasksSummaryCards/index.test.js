import React from 'react';
import { shallow } from 'enzyme';
import * as nhhStyles from 'nhh-styles';
import { TasksSummaryCardsComponent } from './';
import NoDataMessage from '../../components/NoDataMessage';

nhhStyles.formatting.formatDate = jest.fn(() => '10/08/2018 - 8:52pm');

describe('<TasksSummaryCardsComponent />', () => {
  let props;

  beforeEach(() => {
    props = {
      loading: false,
      noDataMsg: 'there is no data sorry',
      resultCount: 2,
      getNextPage: () => {},
      tasks: [
        {
          id: 'foo',
          type: 'manual',
          status: 'Open',
          title: 'this is a title',
          dueDate: '2018-08-27T15:08:44+00:00',
          owner: {
            id: 'DEF456',
            name: 'Jim Bob',
            type: 'systemusers',
          },
          regardingObjectModel: { id: 'bar', type: 'Arrears' },
          assetName: '666 Craven Rd',
          tenantName: 'Dylan Dog',
        },
        {
          id: 'bar',
          type: 'system',
          status: 'Open',
          title: 'this is a title',
          dueDate: '2018-08-27T15:08:44+00:00',
          owner: {
            id: 'ABC123',
            name: 'Fred Bloggs',
            type: 'systemusers',
          },
          regardingObjectModel: { id: 'foo', type: 'Arrears' },
          assetName: 'Beverly Hills',
          tenantName: 'Lennox Lewis',
        },
      ],
      cardLabels: {
        heading: 'Tasks',
        dueDate: 'Due Date',
        taskOwner: 'Task Owner',
        title: 'Title',
        customerName: 'Customer Name',
        address: 'Address',
        status: 'Status',
        overdue: 'overdue!',
      },
      theme: {
        colors: {
          secondary: 'hotpink',
          support: {
            one: 'red',
            two: 'blue',
          },
        },
      },
      userId: 'ABC123',
      youText: 'You',
    };
  });

  it('renders correctly', () => {
    expect(shallow(<TasksSummaryCardsComponent {...props} />)).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const testProps = { ...props, loading: true };
    const el = shallow(<TasksSummaryCardsComponent {...testProps} />);
    expect(el.find(nhhStyles.Loader).length).toBe(1);
  });

  it('renders correctly when there is no data', () => {
    let testProps = { ...props, tasks: [] };
    let el = shallow(<TasksSummaryCardsComponent {...testProps} />);
    expect(el.find(NoDataMessage).length).toBe(1);

    testProps = { ...props, tasks: null };
    el = shallow(<TasksSummaryCardsComponent {...testProps} />);
    expect(el.find(NoDataMessage).length).toBe(1);
  });
});
