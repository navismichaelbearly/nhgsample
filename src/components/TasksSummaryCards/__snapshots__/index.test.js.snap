// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksSummaryCardsComponent /> renders correctly 1`] = `
<l
  items={
    Array [
      Object {
        "assetName": "666 Craven Rd",
        "dueDate": "2018-08-27T15:08:44+00:00",
        "id": "foo",
        "owner": Object {
          "id": "DEF456",
          "name": "Jim Bob",
          "type": "systemusers",
        },
        "regardingObjectModel": Object {
          "id": "bar",
          "type": "Arrears",
        },
        "status": "Open",
        "tenantName": "Dylan Dog",
        "title": "this is a title",
        "type": "manual",
      },
      Object {
        "assetName": "Beverly Hills",
        "dueDate": "2018-08-27T15:08:44+00:00",
        "id": "bar",
        "owner": Object {
          "id": "ABC123",
          "name": "Fred Bloggs",
          "type": "systemusers",
        },
        "regardingObjectModel": Object {
          "id": "foo",
          "type": "Arrears",
        },
        "status": "Open",
        "tenantName": "Lennox Lewis",
        "title": "this is a title",
        "type": "system",
      },
    ]
  }
  onNextPage={[Function]}
  pageSize={6}
  render={[Function]}
  totalNumberOfItems={2}
/>
`;
